import "resource.proto";
import "item.proto";

package SanProto;

//随机宝箱信息
message ChestInfo {
  required int32 next_gap_time = 1;
  required int32 retain_time = 2;
  optional ResourceInfo resource = 3;
  repeated ItemInfo reward_items = 4;

}

//开宝箱
message OpenChestReq {
  repeated ItemInfo items = 1;  //领取红包后的物品信息
}
message OpenChestRes {
  required int32 status = 1;
  optional ChestInfo chest = 2;  //下一次开宝箱的信息
  optional ResourceInfo resource = 3;
}


//查询宝箱
message QueryChestReq {

}
message QueryChestRes {
  required int32 status = 1;
  optional ChestInfo chest = 2;    //下一次宝箱信息
}

